### PR TITLE
Expose additional attributes on the Attribute type

### LINKED ISSUE #12209: Re-introduce members on Attribute
The following members on Attribute need to be brought back:

- [ ] IsDefaultAttribute
- [ ] TypeId
- [ ] Match

### PR DESCRIPTION
**Pull request by AlexGhiondea**
Fixes #12209

/cc @weshaggard

### CODE REVIEW

**Comment by danmosemsft** [comment]
Can we have a test or two..

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -337,6 +337,9 @@ public abstract partial class Attribute
         protected Attribute() { }
         public override bool Equals(object obj) { return default(bool); }
         public override int GetHashCode() { return default(int); }
+        public virtual bool IsDefaultAttribute(){ return default(bool); }
+        public virtual bool Match(object obj){ return default(bool); }
```
Comment (justinvp) [inline src/System.Runtime/ref/System.Runtime.cs]: Nit: Add a space between `)` and `{` in the two lines above.

Comment (AlexGhiondea) [inline src/System.Runtime/ref/System.Runtime.cs]: Done

**Comment by AlexGhiondea** [comment]
@danmosemsft there are no interesting things to tests for those 3 APIs as those methods are meant to be overwritten and do more interesting stuff. 
I added a basic test to cover the default implementation though.

**Comment by AlexGhiondea** [comment]
@dotnet-bot please test Innerloop CentOS7.1 Debug Build and Test (exit code 139)

**Comment by weshaggard** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please

**Pull request merged**